cmake/helpers.cmake

```diff
@@ -135,16 +135,19 @@ macro(add_sanitizer_flag flag)
 
     # Check C and CXX compilers for given sanitizer flag.
     check_c_compiler_flag("${SANITIZER_FLAG}" "C_${check_name}")
-    check_cxx_compiler_flag("${SANITIZER_FLAG}" "CXX_${check_name}")
+    if (CMAKE_CXX_COMPILE_FEATURES)
+        check_cxx_compiler_flag("${SANITIZER_FLAG}" "CXX_${check_name}")
+    endif()
     if (NOT ${C_${check_name}} OR NOT ${CXX_${check_name}})
         message(FATAL_ERROR "${flag} sanitizer is not supported (either by C or CXX compiler)")
     endif()
 
     # Check C and CXX compilers for sanitizer arguments.
     if (${SANITIZER_ARGS})
         check_c_compiler_flag("${SANITIZER_ARGS}" "C_HAS_SAN_ARGS")
-        check_cxx_compiler_flag("${SANITIZER_ARGS}" "CXX_HAS_SAN_ARGS")
-
+        if (CMAKE_CXX_COMPILE_FEATURES)
+            check_cxx_compiler_flag("${SANITIZER_ARGS}" "CXX_HAS_SAN_ARGS")
+        endif()
         if (NOT ${C_HAS_SAN_ARGS} OR NOT ${CXX_HAS_SAN_ARGS})
```

Comment:
This will probably fail when `UMF_BUILD_TESTS` and `UMF_BUILD_LIBUMF_POOL_DISJOINT` options are turned `OFF`.
Could you set the value of `CXX_HAS_SAN_ARGS` to one of the positive constants when CXX compiler should be ignored? (https://cmake.org/cmake/help/latest/command/if.html#constant)

Reply:
hmm.. I think you're on to something... CXX_HAS_SAN_ARGS should be empty here, meaning false. But why didn't it fail on the CI? maybe some values were cached...?

Reply:
It seems to work with both CMake config variables [turned off](https://github.com/oneapi-src/unified-memory-framework/actions/runs/7723525323/job/21053822402?pr=167)

Reply:
Alright, I see that it tests just the C compiler for appropriate sanitizer and doesn't fail, it seems to work correctly.

Reply:
hmm... heh, do we test this part of the code in FastBuild?

Reply:
No, that's why I've [linked](https://github.com/PatKamin/unified-memory-framework/actions/runs/7723653607/job/21054205754) to the modified branch.

Reply:
heh, ok, I'm blind I didn't see your links. Nonetheless, after our call I dug a little into the code and found the issue (issues to be precise) 😄 

Please see my patch here:
https://github.com/lukaszstolarczuk/unified-memory-framework/commit/83c4c331f0947ad195cadbe4c059219dab72246d

If you want some explanations about these, please contact me 😉 

You can see a few of my tries with debug prints, here:
https://github.com/lukaszstolarczuk/unified-memory-framework/actions

Reply:
But it was meant to be an `OR`, I've had a chat with @bratpiorka about this. Couldn't this cause a situation where code compiled with the C compiler is sanitized, while code compiled with the C++ compiler remains unsanitized, or vice versa?

Reply:
heh, while this may be a valid point, questions:
1. when could that happen?
2. sanitizers are mostly for our use case - is it something we could expect in our CI?
3. do you have a quick solution to that `if` to wrap this PR up...? 😄

Reply:
I've updated the PR with a quick solution, should work now as expected.

Reply:
I thought that setting `CXX_HAS_SAN_ARGS` to a default value and leaving the `AND` as is was sufficient.

Reply:
Now, the code looks neat, but TSAN started failing 😄